File: Data Structures & Algorithms/majority-element/submission-0.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        hashmap = {}
        majority = len(nums)//2

        for n in nums:
            hashmap[n] = hashmap.get(n, 0) + 1
        
        for n in hashmap:
            if hashmap[n] > majority:
                return n

                

File: Data Structures & Algorithms/subarray-sum-equals-k/submission-0.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        int result = 0;
        int curSum = 0;

        Map<Integer, Integer> prefixSums = new HashMap<>();
        prefixSums.put(0,1);

        for (int i = 0; i < nums.length; i++) {
            curSum += nums[i];
            int diff = curSum -k;
            result += prefixSums.getOrDefault(diff, 0);
            prefixSums.put(curSum, prefixSums.getOrDefault(curSum, 0) + 1);
        }
        return result;
    }
}